api/configuration/v1alpha1/kongpluginbinding_types.go

```diff
@@ -81,66 +92,57 @@ type KongPluginBindingSpec struct {
 	//
-	// TODO(mlavacca): we need to figure out how to deal with global plugins. By means of this new API,
-	// KongClusterPlugin can be replaced by kongPluginBindings with no Kong references. This way we'd be
-	// more coherent with the Konnect approach.
-	// https://github.com/Kong/kubernetes-configuration/issues/7
-	Kong *KongReferences `json:"kong,omitempty"`
-
-	// TODO(mlavacca): let's defer this one to the future as we are not sure about the shape we want to give it.
-	// https://github.com/Kong/kubernetes-configuration/issues/8
-	// EntityReference        *GenericEntityRef `json:"genericEntityRef,omitempty"`
-}
+	// +optional
+	// +kubebuilder:validation:XValidation:message="group/kind not allowed for the routeRef",rule="(self.kind == 'KongRoute' && self.group == 'configuration.konghq.com') || (self.kind == 'Ingress' && self.group == 'networking.k8s.io') || (self.kind == 'HTTPRoute' && self.group == 'gateway.networking.k8s.io') || (self.kind == 'GCPRoute' && self.group == 'gateway.networking.k8s.io')"
+	RouteReference *TargetRefWithGroupKind `json:"routeRef,omitempty"`
 
-type KongReferences struct {
-	RouteReference         *EntityRef `json:"routeRef,omitempty"`
-	ServiceReference       *EntityRef `json:"serviceRef,omitempty"`
-	ConsumerReference      *EntityRef `json:"consumerRef,omitempty"`
-	ConsumerGroupReference *EntityRef `json:"consumerGroupRef,omitempty"`
+	// +optional
+	// +kubebuilder:validation:XValidation:message="group/kind not allowed for the serviceRef",rule="(self.kind == 'KongService' && self.group == 'configuration.konghq.com') || (self.kind == 'Service' && (self.group == '' || self.group == 'core'))"
+	ServiceReference       *TargetRefWithGroupKind `json:"serviceRef,omitempty"`
+	ConsumerReference      *TargetRef              `json:"consumerRef,omitempty"`
+	ConsumerGroupReference *TargetRef              `json:"consumerGroupRef,omitempty"`
 }
 
 type PluginRef struct {
-	// TODO(mattia): What about cross-namespace references? Do we want to introduce a namespace field
-	// to allow such a reference? We could allow it and require a RefGrant.
+	// TODO(mattia): cross-namespace references are not supported yet.
 	// https://github.com/Kong/kubernetes-configuration/issues/9
 
 	// Name is the name of the KongPlugin or KongClusterPlugin resource.
 	// +kubebuilder:validation:Required
 	Name string `json:"name"`
+
 	// kind can be KongPlugin or KongClusterPlugin. If not set, it is assumed to be KongPlugin.
 	// +kubebuilder:validation:Enum=KongPlugin;KongClusterPlugin
 	// +kubebuilder:default:=KongPlugin
 	Kind *string `json:"kind,omitempty"`
 }
 
-type EntityRef struct {
+type TargetRef struct {
 	// Name is the name of the entity.
 	// +kubebuilder:validation:Required
 	Name string `json:"name"`
 }
 
-// type GenericEntityRef struct {
-// 	// Name is the name of the generic entity.
-// 	// +kubebuilder:validation:Required
-// 	Name string `json:"name"`
+// TODO(mlavacca): add xvalidation between group and kind
```

Comment:
Can we add those in this PR? Or is there anything that's blocking us from adding this now?

Reply:
It's already implemented. This comment was a leftover. Just removed